src/units/abilities.cpp

```diff
@@ -2048,7 +2059,15 @@ effect::effect(const unit_ability_list& list, int def, const_attack_ptr att, boo
 		effect_list_.push_back(val.second);
 	}
 
-	composite_value_ = static_cast<int>((value_set + addition) * multiplier / divisor);
+	if(max_value && min_value) {
+		composite_value_ = std::clamp(*min_value, *max_value, static_cast<int>((value_set + addition) * multiplier / divisor));
```

Comment:
Undefined behavior if `*min_value > *max_value`.

Reply:
i will define what if *min_value > *max_value then composite_value_ not clamped